CMakeLists.txt

```diff
@@ -310,8 +310,21 @@ endif()
 check_function_exists(gethostname HAVE_GETHOSTNAME)
 
 if(WIN32)
-  check_library_exists_concat("ws2_32" getch        HAVE_LIBWS2_32)
-  check_library_exists_concat("winmm"  getch        HAVE_LIBWINMM)
+  check_library_exists_concat("ws2_32" getch HAVE_LIBWS2_32_getch)
+  if(HAVE_LIBWS2_32_getch)
+     set(HAVE_LIBWS2_32 1)
```

Comment:
I'm almost sure this could be done simpler because CMake caches the results and does not run the checks again if it has already found the thing. So correct me if I'm wrong but this should also work in each scenario:
```
check_library_exists_concat("ws2_32" getch        HAVE_LIBWS2_32)
check_library_exists_concat("ws2_32" socket       HAVE_LIBWS2_32)
```

Reply:
I think so, but I don't know how to disable caching results in this case.

Reply:
Why would you?
Unless I misunderstood the issue (judging from the code I didn't) it's perfectly OK because caching only kicks in if the variable is somehow positive (has a value other than "NO", "0", "FALSE", "...-NOTFOUND"). That means if "getch" variant finds it, it's found and the "socket" variant will not really be executed. Otherwise the "socket" variant takes over.